cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1845,15 +1845,13 @@ macro(build_bzip2)
     BZIP2_STATIC_LIB
     "${BZIP2_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}bz2${CMAKE_STATIC_LIBRARY_SUFFIX}")
 
-  set(BZIP2_EXTRA_ARGS "CC=${CMAKE_C_COMPILER}" "CFLAGS=${EP_C_FLAGS}")
```

Comment:
Seems like passing `CC` causes the issue, need to investigate it in more detail.

Reply:
macOS SDK headers are no longer installed under /usr/include so when we use system clang it cannot find its headers unless we explicitly pass the SDK path CFLAGS.